### PR TITLE
Defer sitemap selection until onStart() and refactor pending action handling

### LINKED ISSUE #1430: IllegalStateException: Can not perform this action after onSaveInstanceState
````
Fatal Exception: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=1002, result=-1, data=Intent { (has extras) }} to activity {org.openhab.habdroid.beta/org.openhab.habdroid.ui.MainActivity}: java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
       at android.app.ActivityThread.deliverResults + 4936(ActivityThread.java:4936)
       at android.app.ActivityThread.handleSendResult + 4978(ActivityThread.java:4978)
       at android.app.servertransaction.ActivityResultItem.execute + 49(ActivityResultItem.java:49)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks + 108(TransactionExecutor.java:108)
       at android.app.servertransaction.TransactionExecutor.execute + 68(TransactionExecutor.java:68)
       at android.app.ActivityThread$H.handleMessage + 2123(ActivityThread.java:2123)
       at android.os.Handler.dispatchMessage + 109(Handler.java:109)
       at android.os.Looper.loop + 207(Looper.java:207)
       at android.app.ActivityThread.main + 7470(ActivityThread.java:7470)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run + 524(RuntimeInit.java:524)
       at com.android.internal.os.ZygoteInit.main + 958(ZygoteInit.java:958)
Caused by java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
       at androidx.fragment.app.FragmentManagerImpl.checkStateLoss + 2080(FragmentManager.java:2080)
       at androidx.fragment.app.FragmentManagerImpl.enqueueAction + 2106(FragmentManager.java:2106)
       at androidx.fragment.app.BackStackRecord.commitInternal + 683(BackStackRecord.java:683)
       at androidx.fragment.app.BackStackRecord.commit + 637(BackStackRecord.java:637)
       at org.openhab.habdroid.ui.activity.ContentControllerOnePane.executeStateUpdate$mobile_fullBetaRelease + 57(ContentControllerOnePane.kt:57)
       at org.openhab.habdroid.ui.activity.ContentController.updateFragmentState + 437(ContentController.kt:437)
       at org.openhab.habdroid.ui.activity.ContentController.openSitemap + 182(ContentController.kt:182)
       at org.openhab.habdroid.ui.MainActivity.onActivityResult + 335(MainActivity.kt:335)
       at android.app.Activity.dispatchActivityResult + 7775(Activity.java:7775)
       at android.app.ActivityThread.deliverResults + 4929(ActivityThread.java:4929)
       at android.app.ActivityThread.handleSendResult + 4978(ActivityThread.java:4978)
       at android.app.servertransaction.ActivityResultItem.execute + 49(ActivityResultItem.java:49)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks + 108(TransactionExecutor.java:108)
       at android.app.servertransaction.TransactionExecutor.execute + 68(TransactionExecutor.java:68)
       at android.app.ActivityThread$H.handleMessage + 2123(ActivityThread.java:2123)
       at android.os.Handler.dispatchMessage + 109(Handler.java:109)
       at android.os.Looper.loop + 207(Looper.java:207)
       at android.app.ActivityThread.main + 7470(ActivityThread.java:7470)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run + 524(RuntimeInit.java:524)
       at com.android.internal.os.ZygoteInit.main + 958(ZygoteInit.java:958)
````

Multiple Android versions, multiple devices, Version 2.7.10-beta.

### PR DESCRIPTION
**Pull request by maniac103**
Fixes #1430

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt [hunk 1]
```diff
@@ -704,36 +682,41 @@ class MainActivity : AbstractBaseActivity(), ConnectionFactory.UpdateListener {
...
+    private fun executeActionIfPossible(action: PendingAction): Boolean = when {
+        action is PendingAction.ChooseSitemap && isStarted -> {
+            chooseSitemap()
+            true
+        }
+        action is PendingAction.OpenSitemapUrl && isStarted && serverProperties != null -> {
+            buildUrlAndOpenSitemap(action.url)
+            true
+        }
+        action is PendingAction.OpenHabPanel && isStarted && serverProperties?.hasHabpanelInstalled() == true -> {
```
Comment (mueller-ma) [inline mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt]: `OpenHabPanel` vs `hasHabpanelInstalled`. Can you rename latter to `hasHabPanelInstalled`?

**Comment by mueller-ma** [comment]
There's one valid Stickler issue.
How do you work with AS right now? Doesn't the workplace.xml produce tons of conflicts?

**Comment by maniac103** [comment]
> There's one valid Stickler issue.

Fixed.

> How do you work with AS right now? Doesn't the workplace.xml produce tons of conflicts?

No. It generates a local diff, which is a bit annoying, but it didn't generate conflicts for me yet.

File: mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt [hunk 2]
```diff
@@ -935,8 +910,8 @@ class MainActivity : AbstractBaseActivity(), ConnectionFactory.UpdateListener {
         propsUpdateHandle = null
     }
 
-    private fun manageHabpanelShortcut(visible: Boolean) {
-        manageShortcut(visible, "habpanel", ACTION_HABPANEL_SELECTED,
+    private fun manageHabPanelShortcut(visible: Boolean) {
+        manageShortcut(visible, "habPanel", ACTION_HABPANEL_SELECTED,
```
Comment (mueller-ma) [inline mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt]: If this is changed, the HABPanel shortcut will be duplicated.

Comment (maniac103) [reply mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt]: Oops, this change wasn't intentional.

**Pull request merged**